src/systemctl/systemctl.c

```diff
@@ -2581,7 +2580,22 @@ static int unit_find_paths(
                         }
                 }
 
-                r = set_put(names, basename(path));
+                if (path)
+                        /* We found the unit file. If we followed symlinks, this name might be
+                         * different then the unit_name with started with. Look for dropins matching
+                         * that "final" name. */
+                        r = set_put(names, basename(path));
+                else if (!template)
+                        /* No unit file, let's look for dropins matching the original name.
+                         * systemd has fairly complicated rules (based on unit type and provenience),
```

Comment:
but it's the usual technical term used in cases like this, hence I think it's good the way it is...